tweepy/api.py

```diff
@@ -242,6 +262,70 @@ def media_upload(self, filename, *args, **kwargs):
             upload_api=True
         )(*args, **kwargs)
 
+    def upload_chunked(self, filename, *args, **kwargs):
+        """ :reference https://developer.twitter.com/en/docs/media/upload-media/uploading-media/chunked-media-upload
+            :allowed_param:
+        """
+        f = kwargs.pop('file', None)
+
+        # Media category is dependant on whether media is attached to a tweet
+        # or to a direct message. Assume tweet by default.
+        is_direct_message = kwargs.pop('is_direct_message', False)
+
+        # Initialize upload (Twitter cannot handle videos > 15 MB)
+        headers, post_data, fp = API._chunk_media('init', filename, self.max_size_chunked, form_field='media', f=f, is_direct_message=is_direct_message)
+        kwargs.update({'headers': headers, 'post_data': post_data})
+
+        # Send the INIT request
+        media_info = bind_api(
+            api=self,
+            path='/media/upload.json',
+            method='POST',
+            payload_type='media',
+            allowed_param=[],
+            require_auth=True,
+            upload_api=True
+        )(*args, **kwargs)
+
+        # If a media ID has been generated, we can send the file
+        if media_info.media_id:
+            # default chunk size is 1MB, can be overridden with keyword argument.
+            # minimum chunk size is 16K, which keeps the maximum number of chunks under 999
+            chunk_size = kwargs.pop('chunk_size', 1024 * 1024)
+            chunk_size = max(chunk_size, 16 * 2014)
+
+            fsize = os.path.getsize(filename)
+            nloops = int(fsize / chunk_size) + (1 if fsize % chunk_size > 0 else 0)
+
+            for i in range(nloops):
+                headers, post_data, fp = API._chunk_media('append', filename, self.max_size_chunked, chunk_size=chunk_size, f=fp, media_id=media_info.media_id, segment_index=i, is_direct_message=is_direct_message)
```

Comment:
`self.max_size_chunked` is never defined. Throws errors when I try to test this PR. Should be set accordingly in api.__init__()

Reply:
```
  File "weekly-pitches.py", line 115, in post_tweet
    media_ids = [api.media_upload(video).media_id_string]
  File "/home/marco/weekly-pitches/venv-weekly/lib/python3.8/site-packages/tweepy/api.py", line 245, in media_upload
    return self.upload_chunked(filename, f=f, *args, **kwargs)
  File "/home/marco/weekly-pitches/venv-weekly/lib/python3.8/site-packages/tweepy/api.py", line 276, in upload_chunked
    headers, post_data, fp = API._chunk_media('init', filename, self.max_size_chunked, form_field='media', f=f, is_direct_message=is_direct_message)
AttributeError: 'API' object has no attribute 'max_size_chunked'
```

Reply:
Thanks for testing it out. This is actually an old reference from #929 that hasn't been updated.